owncloud/owncloud.py

```diff
@@ -52,6 +52,140 @@ class HTTPResponseError(ResponseError):
+
+        :returns: name of the share recipient
+        """
+        if 'share_with' in self.share_info:
+            return self.share_info['share_with']
+        return None
+
+    def get_share_with_displayname(self):
+        """Returns the share recipient displayname.
+        If share_with_displayname cannot be returned, None is returned instead
+        :returns: name of the share recipient
+        """
+        if 'share_with_displayname' in self.share_info:
+            return self.share_info['share_with_displayname']
+        return None
+
+    def get_path(self):
+        """Returns the path of the shared file/folder relative to the
+        caller's filesystem.
+
+        :returns: path to the shared file/folder
+        """
+        if 'path' in self.share_info:
+            return self.share_info['path']
+        return None
+
+    def get_permissions(self):
+        """Returns the share permissions.
+        See OCS_PERMISSION_* constants.
+
+        :returns: share permissions
+        """
+        return self.__get_int('permissions')
+
+    def get_share_time(self):
+        """Returns the share time.
+
+        :returns: share timestamp
+        :rtype: datetime object
+        """
+        return datetime.datetime.fromtimestamp(
+            self.__get_int('stime')
+        )
+
+    def get_expiration(self):
+        """Returns the expiration date.
+
+        :returns: expiration date
+        :rtype: datetime object
+        """
+        exp = self.__get_int('expiration')
+        if exp is not None:
+            return datetime.datetime.fromtimestamp(
+                exp
+            )
+        return None
+
+    def get_token(self):
+        if 'token' in self.share_info:
+            return self.share_info['token']
+        return None
+
+    def get_link(self):
+        if 'link' in self.share_info:
+            return self.share_info['link']
+        return None
+
+    def get_uid_owner(self):
+        """Returns the user id of the owner.
+
+        :returns: owner user id
+        """
+        if 'uid_owner' in self.share_info:
+            return self.share_info['uid_owner']
+        return None
+
+    def get_displayname_owner(self):
+        """Returns the display name of the owner.
+
+        :returns: display name of owner
+        """
+        if 'displayname_owner' in self.share_info:
+            return self.share_info['displayname_owner']
+        return None
+
+    def __str__(self):
+        info = ''
+        for k, v in self.share_info.iteritems():
+            info += '%s=%s,' % (k, v)
+        return 'ShareInfo(%s)' % info[:-1]
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __get_int(self, key):
+        """Simple wrapper which converts value to Integer
+           in silently manner (w/o raising exception)"""
+        try:
+            value = int(self.share_info[key])
+            return value
+        except:
+            return None
+
 class PublicShare():
```

Comment:
We can remove UserShare.

But we cannot remove PublicShare because it exists since v0.1, that would break existing API.
This, unless ShareInfo's API is compatible. Goal is to not break scripts that were using v0.1.